langchain/utilities/graphql.py

```diff
@@ -0,0 +1,57 @@
+import json
+from typing import Any, Dict, Optional
+
+from pydantic import BaseModel, Extra, root_validator
+
+
+class GraphQLAPIWrapper(BaseModel):
+    """Wrapper around GraphQL API.
+
+    To use, you should have the ``gql`` python package installed.
+    This wrapper will use the GraphQL API to conduct queries.
+    """
+
+    custom_headers: Optional[Dict[str, str]] = None
+    graphql_endpoint: str
+    gql_client: Any  #: :meta private:
+    gql_function: Any  #: :meta private:
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that the python package exists in the environment."""
+        try:
+            from gql import Client, gql
+            from gql.transport.requests import RequestsHTTPTransport
+
+            headers = values.get("custom_headers", {})
```

Comment:
lines 31-40 can live outside (after) the try/except

Reply:
I left the imports inside the try because they are optional, similar to the pattern used in [this section of the wikipedia utility](https://github.com/hwchase17/langchain/blob/3637d6da6e51cd8b96a9a4f91eb71e8db21a8b20/langchain/utilities/wikipedia.py#L39).